docs/design/architecture.md

```diff
@@ -3,89 +3,103 @@ title: Architecture
 sidebar_position: 1
 ---
 
-WIP on Notion. 
+Quickwit has been designed from the ground up for cost-efficiency, scalability, and cloud-friendly. 
```

Comment:
```suggestion
Quickwit has been designed from the ground up for cost-efficiency and scalability. 
```

We can keep cloud-friendliness, but as is, this is an adjective within a list of nouns